batchlib/mongo/result_writer.py

```diff
@@ -0,0 +1,163 @@
+from batchlib.util import get_logger
+
+ASSAY_ANALYSIS_RESULTS = 'immuno-assay-analysis-results'
+
+logger = get_logger('Workflow.BatchJob.DbResultWriter')
+
+
+def read_table(f, name):
+    key = 'tables/%s' % name
+    g = f[key]
+    ds = g['cells']
+    table = ds[:]
+
+    ds = g['columns']
+    column_names = [col_name.decode('utf-8') for col_name in ds[:]]
+
+    def _col_dtype(column):
+        try:
+            column.astype('int')
+            return 'int'
+        except ValueError:
+            pass
+        try:
+            column.astype('float')
+            return 'float'
+        except ValueError:
+            pass
+        return np.str
+
+    # find the proper dtypes for the columns and cast
+    dtypes = [_col_dtype(col) for col in table.T]
+    columns = [col.astype(dtype) for col, dtype in zip(table.T, dtypes)]
+    n_rows = table.shape[0]
+
+    table = [[col[row] for col in columns] for row in range(n_rows)]
+
+    # a bit hacky, but we use pandas to handle the mixed dataset
+    df = pd.DataFrame(table)
+
+    return column_names, df.values
+
+
+def _get_table_names(f):
+    table_names = set()
+
+    def _visitor(name):
+        if name.endswith('cells'):
+            table_names.add(name[:-6])
+
+    f['tables'].visit(_visitor)
+    return list(table_names)
+
+
+def _table_object(column_names, table):
+    # document attributes cannot contain '.'
+    column_names = [cn.replace('.', '_') for cn in column_names]
+    return [
+        dict(zip(column_names, table_row)) for table_row in table
+    ]
+
+
+def _get_result_tables(in_file):
+    with h5py.File(in_file, 'r') as f:
+        result_tables = {}
+        for table_name in _get_table_names(f):
+            column_names, table = read_table(f, table_name)
+            result_tables[table_name] = _table_object(column_names, table)
+
+        return result_tables
+
+
+def _get_log_path(work_dir):
+    logs = list(glob.glob(os.path.join(work_dir, '*.log')))
+    assert len(logs) == 1
+    return logs[0]
+
+
+def _parse_workflow_duration(work_dir):
+    """
+    Reads workflow duration by parsing the first and the last log event in the log file and taking the time difference
+    """
+    log_path = _get_log_path(work_dir)
+    # TODO: implement
+    return 0
+
+
+def _parse_workflow_name(work_dir):
+    """
+    Parses workflow name from the log file
+    """
+    log_path = _get_log_path(work_dir)
+    logfile = os.path.split(log_path)[1]
+    workflow_name = os.path.splitext(logfile)[0]
+    return workflow_name
+
+
+def _get_git_sha():
+    try:
+        repo = git.Repo(search_parent_directories=True)
+        return repo.head.object.hexsha
+    except Exception as e:
+        logger.warning(f'Cannot get batchlib_version: {e}')
+        return None
+
+
+class DbResultWriter(BatchJobOnContainer):
+    def __init__(self, username, password, host, port=27017, db_name='covid',
+                 input_pattern='*.hdf5', **super_kwargs):
```

Comment:
@constantinpape this might not work as expected: basically I assume that that only `<plate_name>_tables.hdf5` will be passed to the `run()` method, but looking at the `BaseJob.__call__()` semantic this might not be the case. Alternatively we can just get the table hdf5 by:
```
list(glob.glob(self.folder, '*.hdf5'))[0]
```
which is more explicit, but maybe less elegant.